app/src/main/java/com/boosters/promise/ui/notification/AlarmDirector.kt

```diff
@@ -0,0 +1,106 @@
+package com.boosters.promise.ui.notification
+
+import android.app.AlarmManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Context.ALARM_SERVICE
+import android.content.Intent
+import com.boosters.promise.data.alarm.Alarm
+import com.boosters.promise.data.alarm.AlarmRepository
+import com.boosters.promise.data.promise.Promise
+import kotlinx.coroutines.*
+import java.util.*
+
+class AlarmDirector(
+    private val context: Context,
+    private val alarmRepository: AlarmRepository
+) {
+
+    private val coroutineScope by lazy { CoroutineScope(Dispatchers.IO) }
+    private val alarmManager = context.getSystemService(ALARM_SERVICE) as AlarmManager
+
+    fun registerAlarm(promise: Promise) {
+        val date = promise.date.split("/").map { it.toInt() }
+        val time = promise.time.split(":").map { it.toInt() }
+        val cal = Calendar.getInstance()
+        cal.set(Calendar.YEAR, date[0])
+        cal.set(Calendar.MONTH, date[1] - 1)
+        cal.set(Calendar.DAY_OF_MONTH, date[2])
+        cal.set(Calendar.HOUR_OF_DAY, time[0])
+        cal.set(Calendar.MINUTE, time[1])
+
+        if (Calendar.getInstance() >= cal) {
+            coroutineScope.launch {
+                async {
+                    alarmRepository.getAlarm(promise.promiseId)
+                }.await().onSuccess {
+                    alarmRepository.deleteAlarm(promise.promiseId)
+                }
+                cancel()
+            }
```

Comment:
```suggestion
            coroutineScope.launch {
                alarmRepository.getAlarm(promise.promiseId).onSuccess {
                    alarmRepository.deleteAlarm(promise.promiseId)
                }
            }
```
이런식으로 써보는거는 안되나요?
같은 코루틴내에서 그냥 써도 괜찮아보이거든요.

여기 말고도 다른 함수나 파일에도 비슷하게 작성된 코드들이 있던데 이유가 있을까요?